File: light-control/algorithms/white.py
from ._meta import BaseAlgorithm

class Algorithm(BaseAlgorithm):
    def __init__(self, num_pixels, settings) -> None:
        super().__init__(num_pixels, config, settings)
        for i in range(len(self.pixels)):
            pixel = self.pixels[i]
            pixel.hue = 0
            pixel.sat = 0
            pixel.val = 1
            pixel.white = 0

config = {
    "name": "White",
    "options": {},
    "refresh_rate": 2
}

File: light-control-ts/src/algorithms/white.ts
import { BaseAlgorithm, AlgorithmConfig } from './_meta/index.js'

export const config: AlgorithmConfig = {
  name: 'White',
  options: {},
  refresh_rate: 2,
}

export class Algorithm extends BaseAlgorithm {
  constructor(numPixels: number, settings: Record<string, unknown>) {
    super(numPixels, config, settings)
    for (const pixel of this.pixels) {
      pixel.hue = 0
      pixel.sat = 0
      pixel.val = 1
      pixel.white = 0
    }
  }
}
